connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+                scroll_url = f"{url}&next_page_token={next_page_token}"
+            else:
+                break
+
+
+class ZoomClient:
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self._logger = logger
+
+        self.configuration = configuration
+        self.http_session = aiohttp.ClientSession(raise_for_status=True)
+
+        self.api_token = ZoomAPIToken(
+            http_session=self.http_session,
+            configuration=configuration,
+            logger_=self._logger,
+        )
+        self.api_client = ZoomAPISession(
+            http_session=self.http_session,
+            api_token=self.api_token,
+            logger_=self._logger,
+        )
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+        self.api_token.set_logger(self._logger)
+        self.api_client.set_logger(self._logger)
+
+    async def close(self):
+        await self.http_session.close()
+        self.api_client.close()
+
+    async def get_users(self):
+        url = APIS["USERS"].format(base_url=BASE_URL, page_size=MEETING_PAGE_SIZE)
+        async for users in self.api_client.scroll(url=url):
+            for user in users.get("users", []) or []:
+                yield user
+
+    async def get_meetings(self, user_id, meeting_type):
+        url = APIS["MEETINGS"].format(
+            base_url=BASE_URL,
+            user_id=user_id,
+            meeting_type=meeting_type,
+            page_size=MEETING_PAGE_SIZE,
+        )
+        async for meetings in self.api_client.scroll(url=url):
+            for meeting in meetings.get("meetings", []) or []:
+                yield meeting
+
+    async def get_past_meeting(self, meeting_id):
+        url = APIS["PAST_MEETING"].format(base_url=BASE_URL, meeting_id=meeting_id)
+        return await self.api_client.fetch(url=url)
+
+    async def get_past_meeting_participants(self, meeting_id):
+        url = APIS["PAST_MEETING_PARTICIPANT"].format(
+            base_url=BASE_URL, meeting_id=meeting_id, page_size=MEETING_PAGE_SIZE
+        )
+        async for participants in self.api_client.scroll(url=url):
+            for participant in participants.get("participants", []) or []:
+                yield participant
+
+    async def get_recordings(self, user_id):
+        # Zoom recording does not retrieve data that is more than 4 months old.
+        now = datetime.utcnow()
+        for _ in range(4):
+            past_duration = now + timedelta(days=-30)
+            url = APIS["RECORDING"].format(
+                base_url=BASE_URL,
+                user_id=user_id,
+                date_from=format_recording_date(date=past_duration),
+                date_to=format_recording_date(date=now),
+                page_size=MEETING_PAGE_SIZE,
+            )
+            async for recordings in self.api_client.scroll(url=url):
+                for recording in recordings.get("meetings", []) or []:
```

Comment:
would recordings ever have:
```
{
  "meetings": None
}
```
?

If not,  the `or []` feels redundant.

Reply:
Yes faced for multiple objects which returns `None` so kept `or []`